File: Data Structures & Algorithms/binary-tree-diameter/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int result = 0;
    int heightOfTreeNode(TreeNode* root) {
        if (!root) return 0;

        int left = heightOfTreeNode(root->left);
        int right = heightOfTreeNode(root->right);

        result = max(result, (left + right));

        return 1 + max(left, right);
    }
    int diameterOfBinaryTree(TreeNode* root) {
        if (!root) return 0;

        heightOfTreeNode(root);

        return result;
    }
};

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-1.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l, r = 0, 0
        counter = 0
        result = 0
        while r < len(nums):
            if nums[r] == 0:
                counter += 1
            if counter > k:
                if nums[l] == 0:
                    counter -= 1
                l += 1
            result = max(result, r - l + 1)
            r += 1
        return result